appium/webdriver/extensions/gsm.py

```diff
@@ -32,14 +32,9 @@ def set_gsm_signal(self, strength):
         :Args:
          - strength: Signal strength. Can be set Gsm.NONE_OR_UNKNOWN/POOR/MODERATE/GOOD/GREAT
 
-        :Raises:
-        - TypeError - Raises a TypeError if the argument is out of range
-
         :Usage:
             self.driver.set_gsm_signal(Gsm.GOOD)
         """
-        if strength not in [self.NONE_OR_UNKNOWN, self.POOR, self.MODERATE, self.GOOD, self.GREAT]:
```

Comment:
Here is pretty good to show users which args can use for this method.
(Here is client-side logic matter)

Reply:
Added warning log instead of throwing exception.
Here is behavior.

```
➜  mobile git:(master) ✗ python SampleAndroidTest.py
test_01_test_name (__main__.SimpleTest) ... 100 is out of range. Use the value like Gsm.GOOD for signal strength.
ERROR

======================================================================
ERROR: test_01_test_name (__main__.SimpleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "SampleAndroidTest.py", line 48, in test_01_test_name
    self.driver.set_gsm_signal(100)
  File "/Users/atsushimori/.pyenv/versions/3.7.1/lib/python3.7/site-packages/appium/webdriver/extensions/gsm.py", line 41, in set_gsm_signal
    self.execute(Command.SET_GSM_SIGNAL, {'signalStrength': strength, 'signalStrengh': strength})
  File "/Users/atsushimori/.pyenv/versions/3.7.1/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/Users/atsushimori/.pyenv/versions/3.7.1/lib/python3.7/site-packages/appium/webdriver/errorhandler.py", line 29, in check_response
    raise wde
  File "/Users/atsushimori/.pyenv/versions/3.7.1/lib/python3.7/site-packages/appium/webdriver/errorhandler.py", line 24, in check_response
    super(MobileErrorHandler, self).check_response(response)
  File "/Users/atsushimori/.pyenv/versions/3.7.1/lib/python3.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: An unknown server-side error occurred while processing the command. Original error: Invalid signal strength param 100. Supported values: 0,1,2,3,4


----------------------------------------------------------------------
Ran 1 test in 20.444s

FAILED (errors=1)
```

Reply:
what about something like `{name for name, value in vars(self).items() if name.isupper()}` to show the consts instead of the number?

Reply:
I misunderstood for https://github.com/appium/python-client/pull/357/commits/1ca58f4e7895b870c4aedd8f707fe8e5a1d0e65a

> selenium.common.exceptions.WebDriverException: Message: An unknown server-side error occurred while processing the command. Original error: Invalid signal strength param 100. Supported values: 0,1,2,3,4

This log comes from https://github.com/appium/appium-adb/blob/527c8b417d7b62717ececfdf068309b0ee4dbd8f/lib/tools/adb-emu-commands.js#L168 .
So can't handle from client side.

Added log is here.
```
➜  mobile git:(master) ✗ python SampleAndroidTest.py                                                                                                           
test_01_test_name (__main__.SimpleTest) ... 100 is out of range. Use the value in ['Gsm.NONE_OR_UNKNOWN', 'Gsm.POOR', 'Gsm.MODERATE', 'Gsm.GOOD', 'Gsm.GREAT'].
ERROR
(--snip--)
```

Reply:
Ah, okay. I misunderstood it was the warning in this pr